Plugins/ServerPlugin.cpp

```diff
@@ -118,9 +119,13 @@ struct SystemReader : public AsyncReadWorker<SystemType> {
 
 struct CompileReader : public AsyncReadWorker<CompileReply> {
   virtual ~CompileReader() {}
+  Utils::AnsiEscapeCodeHandler ansiHandler;
   virtual void process(const CompileReply& reply) final {
     for (auto log : reply.message()) {
-      emit LogOutput(log.message().c_str());
+	  QList<Utils::FormattedText> txts = ansiHandler.parseText(QString::fromStdString(log.message() + "\n"));
+	  for (auto& str : txts) {
+	    emit LogOutput(str.text, str.format);
+	  }

```

Comment:
This ANSI escape stuff itself could have been made a plugin and more loosely coupled but I suppose that might be overkill.

Reply:
Actually I guess this is fine because now all plugins have ability to log formatted output of their own.